lib/create-plugin.js

```diff
@@ -175,9 +171,9 @@ function createPlugin(api, config = {}) {
               // the attributes to resolver segments as they help
               // inform performance impacts.
               for (const segment of [operationSegment, resolverSegment]) {
-                for (const [key, value] of Object.entries(args)) {
+                for (const [key, value] of Object.entries(flattenedArgs)) {
```

Comment:
this was actually a bug and now I realized I forgot to add a test demonstrating this

Reply:
I'm interested in seeing that bug, please add the test.